### ISSUE TITLE
testing a 'folder of files' locally.

### ISSUE THREAD

**Issue by sturobson** [garris/BackstopJS#191]
Is it possible to test 'everything' from the url provide? For example I have a load of design patterns that can be accessed via `http://localhost:3000/components/preview/` ... currently I need to add the pattern i want to test rather than being able to test everything all at once ... is this possible?

**Comment by garris** [comment]
@sturobson Thanks for posting!   I'm sorry but I am not sure I understand your question -- could you maybe post an example?

**Comment by sturobson** [comment]
@garris sorry, let me try again :) 
Is it possible to use this on multiple pages at once, or is it restricted to one .html file at a time? I've a large pattern library and was wondering if it was possible to point BackstopJS at the folder of patterns for it run through. Does that make better sense? Thanks for replying :)

**Comment by garris** [comment]
Ok. I understand.  Yes.  You can do this using [a module based config file](https://github.com/garris/BackstopJS#using-a-js-based-config-file-version-100).

Instead of hard-coding your config, you would create a three step process. 
-- Step 1: use node FS to look inside a target directory and create an array of file paths you want tested.
-- Step 2: create a new scenarios array by cloning a base scenario object for each file path you created in step 1, replacing the url property in each newly cloned scenario object.
-- Step 3: return a full config object which includes the scenarios array you created in step 2.

Just remember to run `npm run reference` when you create new files to keep everything in sync.

Hope this helps.  Please post back if you complete this script -- I have had requests for this kind of thing before so, I know others would appreciate the example.

Cheers.

**Comment by mathesond2** [comment]
@sturobson did you complete this script? Would love to take a look..thanks!

**Comment by sturobson** [comment]
@mathesond2 @garris I'm looking at this again - I'll try and work out how to grab scenarios per pattern with the ideas above - I'll add the code here if I work it out.

**Comment by garris** [comment]
@sturobson @mathesond2 Thanks for your interest in BackstopJS.  I wanted to let you know there are some recent examples which might help you guys move forward...

1. Simple JS based config...
https://github.com/garris/BackstopJS/tree/master/examples/jsBasedConfig

2. A node app which imports BackstopJS and executes everything programmatically...
https://github.com/garris/BackstopJS/tree/master/examples/nodeIntegration

3. Also, there is a cool project by @danreeves that may be a good starting point as well. This app crawls your site and generates a config file for urls x-links deep.   I think this one is great work -- it would be even cooler to extend this by importing BackstopJS (vis-a-vis example 2 above) and then execute BackstopJS directly by injecting newly-crawled or cached scenarios.  (If that makes sense.)
https://github.com/fffunction/backstop-crawl

**Comment by sturobson** [comment]
Thanks for your pointers @garris - I've managed to get it working in my starter project that uses Fractal ... https://github.com/sturobson/SCL

The config file is here - https://github.com/sturobson/SCL/blob/master/backstopConfig.js and I'm running all the things via gulp - https://github.com/sturobson/SCL/blob/master/gulpfile.js

I'm going to close this now - I'm hoping to do a quick write-up about this too.

**Comment by garris** [comment]
Wow - what a beautiful implementation!  Please send a link when you write it up -- I'd be happy to promote it in the docs!  Cheers!
